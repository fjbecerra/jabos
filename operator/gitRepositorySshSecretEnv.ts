export default function (ssh): any[] {
  return !ssh ? [] : [
    {
      "name": "SSH_PASSPHRASE",
      "valueFrom": {
        "secretKeyRef": {
          "name": ssh.secret,
          "key": ssh.passphrase
        }
      }
    },
    {
      "name": "SSH_KEY",
      "valueFrom": {
        "secretKeyRef": {
          "name": ssh.secret,
          "key": ssh.key
        }
      }
    }
  ]
}