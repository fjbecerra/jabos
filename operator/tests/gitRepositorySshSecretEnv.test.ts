import gitRepositorySshSecretEnv from '../gitRepositorySshSecretEnv';
import { setMock, clearMock } from './settingsMock';

beforeEach(() => {
  setMock();
});

afterEach(() => {
  clearMock();
});

test('gitRepositorySshSecretEnv empty', () => {
  expect(gitRepositorySshSecretEnv(undefined)).toMatchSnapshot();
});

test('gitRepositorySshSecretEnv', () => {
  expect(gitRepositorySshSecretEnv({
    secret: 'secret_value',
    passphrase: 'passphrase_value',
    key: 'key_value'
  })).toMatchSnapshot();
});
