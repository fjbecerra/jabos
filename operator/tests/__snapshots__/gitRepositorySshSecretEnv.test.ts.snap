// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gitRepositorySshSecretEnv 1`] = `
Array [
  Object {
    "name": "SSH_PASSPHRASE",
    "valueFrom": Object {
      "secretKeyRef": Object {
        "key": "passphrase_value",
        "name": "secret_value",
      },
    },
  },
  Object {
    "name": "SSH_KEY",
    "valueFrom": Object {
      "secretKeyRef": Object {
        "key": "key_value",
        "name": "secret_value",
      },
    },
  },
]
`;

exports[`gitRepositorySshSecretEnv empty 1`] = `Array []`;
